examples_tests/39.DenoiserTonemapper/main.cpp

```diff
@@ -586,152 +584,159 @@ void main()
-					shaderConstants.tonemapperParams[0] = tp.keyAndLinearExposure;
-					shaderConstants.tonemapperParams[1] = tp.rcpWhite2;
-					break;
-				}
-				case ToneMapperClass::EO_ACES:
-				{
-					auto tp = ToneMapperClass::Params_t<ToneMapperClass::EO_ACES>(optiXIntensityKeyCompensation,key,extraParam);
-					shaderConstants.tonemapperParams[0] = tp.gamma;
-					shaderConstants.tonemapperParams[1] = (&tp.gamma)[1];
-					break;
-				}
-				default:
+			case ToneMapperClass::EO_REINHARD:
+			{
+				auto tp = ToneMapperClass::Params_t<ToneMapperClass::EO_REINHARD>(optiXIntensityKeyCompensation, key, extraParam);
+				shaderConstants.tonemapperParams[0] = tp.keyAndLinearExposure;
+				shaderConstants.tonemapperParams[1] = tp.rcpWhite2;
+				break;
+			}
+			case ToneMapperClass::EO_ACES:
+			{
+				auto tp = ToneMapperClass::Params_t<ToneMapperClass::EO_ACES>(optiXIntensityKeyCompensation, key, extraParam);
+				shaderConstants.tonemapperParams[0] = tp.gamma;
+				shaderConstants.tonemapperParams[1] = (&tp.gamma)[1];
+				break;
+			}
+			default:
+			{
+				if (core::isnan(key))
 				{
-					if (core::isnan(key))
-					{
-						shaderConstants.tonemapperParams[0] = 0.18;
-						shaderConstants.autoexposureOff = 1u;
-					}
-					else
-						shaderConstants.tonemapperParams[0] = key;
-					shaderConstants.tonemapperParams[0] *= exp2(optiXIntensityKeyCompensation);
-					shaderConstants.tonemapperParams[1] = core::nan<float>();
-					break;
+					shaderConstants.tonemapperParams[0] = 0.18;
+					shaderConstants.autoexposureOff = 1u;
 				}
+				else
+					shaderConstants.tonemapperParams[0] = key;
+				shaderConstants.tonemapperParams[0] *= exp2(optiXIntensityKeyCompensation);
+				shaderConstants.tonemapperParams[1] = core::nan<float>();
+				break;
 			}
-			auto totalSampleCount = param.width*param.height;
-			shaderConstants.percentileRange[0] = lowerPercentile*float(totalSampleCount);
-			shaderConstants.percentileRange[1] = upperPercentile*float(totalSampleCount);
+			}
+			auto totalSampleCount = param.width * param.height;
+			shaderConstants.percentileRange[0] = lowerPercentile * float(totalSampleCount);
+			shaderConstants.percentileRange[1] = upperPercentile * float(totalSampleCount);
 			shaderConstants.normalMatrix = cameraTransformBundle[i].value();
 		}
 
 		// upload image channels and register their buffer
 		{
 			asset::ICPUBuffer* buffersToUpload[EII_COUNT];
-			for (uint32_t j=0u; j<denoiserInputCount; j++)
+			for (uint32_t j = 0u; j < denoiserInputCount; j++)
 				buffersToUpload[j] = param.image[j]->getBuffer();
-			auto gpubuffers = driver->getGPUObjectsFromAssets(buffersToUpload,buffersToUpload+denoiserInputCount,&assetConverter);
+			auto gpubuffers = driver->getGPUObjectsFromAssets(buffersToUpload, buffersToUpload + denoiserInputCount, &assetConverter);
 
 			imagePixelBuffer = core::smart_refctd_ptr<IGPUBuffer>(gpubuffers->operator[](0)->getBuffer());
```

Comment:
after changing tile size to 1024 by 1024, images are skipped
`
While executing a kernel, the device encountered a load or store instruction on an invalid memory address. This leaves the process in an inconsistent state and any further CUDA work will return the same error. To continue using CUDA, the process must be terminated and relaunched.
`